### PR TITLE
Adding nimbus-build-system

### PR DESCRIPTION
**Pull request by richard-ramos**
This changes the build file from nimble to nimbus-build-system, and currently compiles nim and all the dependencies from the `vendor/`. These dependencies are now submodules, so when cloning the repo from scratch
```
git clone https://github.com/status-im/nim-status-client/ --recurse-submodules
```

If you already cloned the repo
```
git submodule update --init --recursive.
```

New targets are added to the makefile:
`make build-linux`
`make build-macos`

Also, `make` executed without targets will determine the OS and build according to it.

To use the nim version installed in your OS instead of the one built with this PR, 
```
make USE_SYSTEM_NIM=1
```

Also setting up the  LD_LIBRARY_PATH changed too
```
export LD_LIBRARY_PATH=vendor/DOtherSide/build/lib/
```

### CODE REVIEW

**Comment by michaelsbradleyjr** [comment]
We don't need to keep [libstatus.h] in `lib/`, can just delete it and add a `.gitkeep` if we want the directory to stick around.

**Comment by michaelsbradleyjr** [comment]
Should [`.nimble/pkgs`](https://github.com/status-im/nim-status-client/tree/build/vendor/.nimble/pkgs) be ignored by git?

I don't see such a directory in [nimbus-build-system/tree/master/vendor](https://github.com/status-im/nimbus-build-system/tree/master/vendor).

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,83 @@
...
+all: $(DEFAULT_TARGET)
+
+# must be included after the default target
+-include $(BUILD_SYSTEM_DIR)/makefiles/targets.mk
+
+deps: | deps-common
+
+update: | update-common
+
+DOTHERSIDE := vendor/DOtherSide/build/lib/libDOtherSideStatic.a
```
Comment (michaelsbradleyjr) [inline Makefile:62]: Can we do the same thing for e.g. `STATUSGO`, having status-go be a submodule like the others in vendor?

Comment (richard-ramos) [reply Makefile:62]: Yes, currently doing that :)

**Comment by richard-ramos** [comment]
> Should [`.nimble/pkgs`](https://github.com/status-im/nim-status-client/tree/build/vendor/.nimble/pkgs) be ignored by git?
> 
> I don't see such a directory in [nimbus-build-system/tree/master/vendor](https://github.com/status-im/nimbus-build-system/tree/master/vendor).

Probably not needed. Will test without it

**Comment by michaelsbradleyjr** [comment]
> Probably not needed. Will test without it

I think it may be needed, but I'm not sure it should be tracked in the repo, if that makes sense. Notice that the files in there are related to paths that are specific to your machine.

**Comment by arnetheduck** [comment]
ping @stefantalpalaru

**Comment by stefantalpalaru** [comment]
Yes, you want "vendor/.nimble" in .gitignore: https://github.com/status-im/nim-beacon-chain/blob/8b4065bc19b56940c7af021650a7388aa7ed46c3/.gitignore#L7

File: README.md [hunk 2]
```diff
@@ -22,59 +22,34 @@ export PATH=$PATH:/path/to/Qt/5.14.2/gcc_64/bin
...
-cmake ..
-make
-```
-
-### 4. Setup Library Path
-
+### 3. Clone the repo
 ```
-export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/path/to/dotherside/build/lib
+git clone https://github.com/status-im/nim-status-client/ --recurse-submodules
```
Comment (stefantalpalaru) [inline README.md:27]: The need to fetch/update the submodules while cloning the project was mitigated by the special default target in the Makefile that does that for you the first time you run `make` with the submodules missing.

**Comment by emizzle** [comment]
> We don't need to keep [libstatus.h](https://github.com/status-im/nim-status-client/blob/build/lib/libstatus.h) in `lib/`, can just delete it and add a `.gitkeep` if we want the directory to stick around.

We probably don't need to keep `lib/` around any longer, right? It was meant for pre-compiled binaries, and we are now using `vendor/` for that.

**Pull request merged**